### PR TITLE
Authentication by proxy

### PR DESCRIPTION
**Pull request by stephane-martin**
Hello,

a simple modification to delegate authentication to the upstream reverse proxy.

When in config `AUTH_PROXY = True`, Realms will look in the HTTP request for header `REMOTE_USER` (classically used by HTTP servers and SSO). If `REMOTE_USER` is given, the user is automatically logged in Realms.

I added a bit of documentation in README too.

Please review.

Thanks,
Stephane
(and I lost 30 minutes to understand that in flask 0.11 is_authenticated is now a property ;))

### CODE REVIEW

File: realms/__init__.py [hunk 1]
```diff
@@ -215,6 +216,23 @@ def root():
...
+            remote_user = request.headers.get(app.config["AUTH_PROXY_HEADER_NAME"])
+            if remote_user:
+                if current_user.is_authenticated:
+                    if current_user.id == remote_user:
+                        return
+                    logger.info("login in realms and login by proxy are different: '{}'/'{}'".format(
+                        current_user.id, remote_user))
+                    logout_user()
+                logger.info("User logged in by proxy as '{}'".format(remote_user))
+                ProxyUser.do_login(remote_user)
```
Comment (gazpachoking) [inline realms/__init__.py]: Shouldn't this bit be in `modules/auth/proxy/hooks.py`?

**Comment by gazpachoking** [comment]
> (and I lost 30 minutes to understand that in flask 0.11 is_authenticated is now a property ;))

Oops, sorry about that. PR merge order added some older flask-login stuff back in there. Fixed now.

File: realms/__init__.py (hunk 1 above)
Comment (stephane-martin) [inline realms/__init__.py]: Sure, done.

**Comment by stephane-martin** [comment]
rebased

**Comment by stephane-martin** [comment]
I think I'm doing something wrong with that rebasing stuff... I keep seing changes about the "is_anonymous" vs "is_anonymous()" issue :-/

**Pull request merged**